lib/matchers/isoDate.js

```diff
@@ -1,7 +1,7 @@
 var _ = require('lodash');
 var factory = require('../factory');
 
-var dateTimeRegex   = /^\d\d\d\d-\d\d-\d\dT\d\d:\d\d:\d\d(.\d\d\d)?(Z|[+-]\d\d:\d\d)?$/;
+var dateTimeRegex   = /^\d\d\d\d-\d\d-\d\dT\d\d:\d\d:\d\d(.\d+)?(Z|[+-]\d\d:\d\d)?$/;
```

Comment:
this can be written a bit more clean as `^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(.\d{3,4})?(Z|[+-]\d{2}:\d{2})?$`